### PR TITLE
Do not use ocramius/package-versions

### PR DESCRIPTION
**Pull request by kubawerlos**
This PR:

- removes `ocramius/package-versions` from dependencies
- starts using `composer-runtime-api` as [suggested](https://github.com/composer/composer/issues/10034#issuecomment-901888729) by @Seldaek

### CODE REVIEW

File: src/Process/Factory/InitialTestsRunProcessFactory.php [hunk 1]
```diff
@@ -79,7 +79,7 @@ public function createProcess(
 
         $process->setTimeout(null); // Ignore the default timeout of 60 seconds
 
-        if (method_exists($process, 'inheritEnvironmentVariables') && version_compare(Versions::getVersion('symfony/console'), 'v4.4', '<')) {
+        if (method_exists($process, 'inheritEnvironmentVariables') && version_compare((string) InstalledVersions::getReference('symfony/console'), 'v4.4', '<')) {
```
Comment (Seldaek) [inline src/Process/Factory/InitialTestsRunProcessFactory.php:82]: Should be getVersion

File: src/Console/Application.php [hunk 2]
```diff
@@ -73,7 +73,7 @@ final class Application extends BaseApplication
     public function __construct(Container $container)
     {
         try {
-            $version = Versions::getVersion(self::PACKAGE_NAME);
+            $version = (string) InstalledVersions::getReference(self::PACKAGE_NAME);
```
Comment (Seldaek) [inline src/Console/Application.php:76]: Should be getVersion

File: src/Process/Factory/MutantProcessFactory.php [hunk 3]
```diff
@@ -84,7 +84,7 @@ public function createProcessForMutant(Mutant $mutant, string $testFrameworkExtr
 
         $process->setTimeout($this->timeout);
 
-        if (method_exists($process, 'inheritEnvironmentVariables') && version_compare(Versions::getVersion('symfony/console'), 'v4.4', '<')) {
+        if (method_exists($process, 'inheritEnvironmentVariables') && version_compare((string) InstalledVersions::getReference('symfony/console'), 'v4.4', '<')) {
```
Comment (Seldaek) [inline src/Process/Factory/MutantProcessFactory.php:87]: Should be getVersion

**Review by Seldaek — CHANGES_REQUESTED** [review]

File: src/Command/ConfigureCommand.php [hunk 4]
```diff
@@ -243,9 +242,9 @@ private function getJsonSchemaPathOrUrl(): string
         }
 
         try {
-            $version = strstr(Versions::getVersion(Application::PACKAGE_NAME), '@', true);
+            $version = InstalledVersions::getReference(Application::PACKAGE_NAME);
```
Comment (maks-rafalko) [inline src/Command/ConfigureCommand.php:245]: seems like this breaks the tests

**Review by maks-rafalko — APPROVED** [review]
Thank you @kubawerlos

**Comment by maks-rafalko** [comment]
compiled PHAR now fails with

```
PHP Fatal error:  Uncaught Error: Class "_HumbugBox11d54ac50db9\Composer\InstalledVersions" not found in phar:///home/maksrafalko/apps/infection/build/infection.phar/src/Console/Application.php:36
Stack trace:
#0 phar:///home/maksrafalko/apps/infection/build/infection.phar/bin/infection(71): _HumbugBox11d54ac50db9\Infection\Console\Application->__construct()
#1 /home/maksrafalko/apps/infection/build/infection.phar(14): require('phar:///home/ma...')
#2 {main}
  thrown in phar:///home/maksrafalko/apps/infection/build/infection.phar/src/Console/Application.php on line 36

```

**Comment by maks-rafalko** [comment]
Reverted due to fatal error above

**Pull request merged**